sql/mysqld.cc

```diff
@@ -8383,6 +8383,13 @@ int mysqld_main(int argc, char **argv)
     if (mysql_bin_log.write_event_to_binlog_and_sync(&prev_gtids_ev))
       unireg_abort(MYSQLD_ABORT_EXIT);
 
+#ifdef WITH_WSREP
+  /* In wsrep_recovery mode, PXC avoid creation of new binlog file for
+  the reason mentioned above. In light of the said flow avoid purge
```

Comment:
The reason is not mentioned above for explaining this behavior.

I think we should mention the resaon here. (Copied from an already existing comment)
```
    /* Don't spawn a new binlog file during wsrep-recovery. Why ?
    - Recovery flow is only going to read existing wsrep saved co-ordinate
      from sys_header. No other action is performed that needs binlogging.
  
    - Existing server flow looks at last binlog to set gtid_executed if server
      was shutdown abruptly. Newly create binlog will not have any such
      information and so restart post wsrep_recover will result in gtid_executed
      to be empty. */
```

Reply:
I'm not sure if it's a good idea to duplicate large comments blocks, but I added it here.

Reply:
I agree. But in this case, having a comment that says 
```
In wsrep_recovery mode, PXC avoid creation of new binlog file for
  the reason mentioned above.
```
without the actual reason above didn't make sense to me.